### ISSUE TITLE
Class expressions handled wrong by REPL now

### ISSUE THREAD

**Issue by takikawa** [racket/typed-racket#220]
This is a recent REPL regression in TR. (note lack of checking and lack of type printout)

````
$ racket -I typed/racket
Welcome to Racket v6.2.900.17.
-> (class object% (super-new))
#<class:readline-input:1:0>
-> (class object% (super-new) (+ 1 "foo"))
#<class:readline-input:2:0>
````

Probably related to recent top-level refactoring since it works correctly in modules. Not a critical issue since it works fine in definitions or other expressions (e.g., wrap a `new` around it).

**Comment by dfeltey** [comment]
There's a similar issue with units at REPL (issue #199), when I looked at this before it seemed like the syntax properties that should be matched against at https://github.com/racket/typed-racket/blob/master/typed-racket-lib/typed-racket/typecheck/tc-toplevel.rkt#L225  were being dropped or ignored leading to expressions not typechecking correctly at the toplevel.

**Comment by takikawa** [comment]
Fixed by 5fa40de5468c3e775b0a3ebe2fb7fcf0f69a56fe
